pkg/build/prune/resolvers.go

```diff
@@ -98,30 +98,34 @@ func (o *perBuildConfigResolver) Resolve() ([]*buildapi.Build, error) {
 	completeStates := util.NewStringSet(string(buildapi.BuildPhaseComplete))
 	failedStates := util.NewStringSet(string(buildapi.BuildPhaseFailed), string(buildapi.BuildPhaseError), string(buildapi.BuildPhaseCancelled))
 
-	results := []*buildapi.Build{}
+	prunableBuilds := []buildapi.Build{}
 	for _, buildConfig := range buildConfigs {
 		builds, err := o.dataSet.ListBuildsByBuildConfig(buildConfig)
 		if err != nil {
 			return nil, err
 		}
 
-		completeBuilds, failedBuilds := []*buildapi.Build{}, []*buildapi.Build{}
+		completeBuilds, failedBuilds := []buildapi.Build{}, []buildapi.Build{}
 		for _, build := range builds {
 			if completeStates.Has(string(build.Status.Phase)) {
-				completeBuilds = append(completeBuilds, build)
+				completeBuilds = append(completeBuilds, *build)
 			} else if failedStates.Has(string(build.Status.Phase)) {
-				failedBuilds = append(failedBuilds, build)
+				failedBuilds = append(failedBuilds, *build)
 			}
 		}
-		sort.Sort(sortableBuilds(completeBuilds))
-		sort.Sort(sortableBuilds(failedBuilds))
+		sort.Sort(sort.Reverse(buildapi.ByCreationTimestamp(completeBuilds)))
```

Comment:
all of these uses assume `buildapi.ByCreationTimestamp` sorts in place... that's probably not a great assumption